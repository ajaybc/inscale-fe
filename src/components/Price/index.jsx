import React, {Fragment} from 'react';

const THOUSAND = 1000;
const MILLION = 1000000;

function formatPrice(price) {
  if (price > MILLION) {
    return `${(price / MILLION)}M USD`
  } else if (price > THOUSAND) {
    return `${(price / THOUSAND)}K USD`
  } else {
    return `${price} USD`
  }
}

export default ({children}) => {
  return <Fragment>
    {formatPrice(children)}
  </Fragment>
} 