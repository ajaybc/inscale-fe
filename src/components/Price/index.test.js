import React from 'react';
import renderer from 'react-test-renderer';

import Price from './index.jsx';

it('Price renders correctly when a value of 2000 is provided', () => {
  const tree = renderer
    .create(<Price>2000</Price>)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('Price renders correctly when a very large value is provided', () => {
  const tree = renderer
    .create(<Price>99999999999</Price>)
    .toJSON();
  expect(tree).toMatchSnapshot();
});