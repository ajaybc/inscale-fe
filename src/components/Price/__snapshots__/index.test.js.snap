// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Price renders correctly when a value of 2000 is provided 1`] = `"2K USD"`;

exports[`Price renders correctly when a very large value is provided 1`] = `"99999.999999M USD"`;
